spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/EntityPropertyValueProvider.java

```diff
@@ -50,7 +53,10 @@
 			return null;
 		}
 		try {
-
+			if (persistentProperty.isEmbedded()) {
```

Comment:
I see you added a  reference to the converter in  this class to do this `read(...)` call. However, when looping through the properties in the converter could you check for if the property is embedded there instead and do the recursive call right there? Perhaps it'll make the changes to the constructors go away, and might be simpler?

Reply:
That's not going to work, because it's not the only place where EntityPropertyValueProvider is used.

Reply:
Other than in tests, Intellij is saying the only place `EntityPropertyValueProvider` is used is in `read(...)` in `DefaultDatastoreEntityConverter`. Doesn't that mean you can just call `read` directly from `read` in the Converter?